sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1260,6 +1263,59 @@ bool _pi_queue::isBatchingAllowed() {
   return (this->QueueBatchSize > 0 && ((ZeSerialize & ZeSerializeBlock) == 0));
 }
 
+pi_result _pi_queue::getOrCreateCopyCommandQueue(
+    int Index, ze_command_queue_handle_t &ZeCopyCommandQueue) {
+  ZeCopyCommandQueue = nullptr;
+
+  // Make sure 'Index' is within limits
+  PI_ASSERT((Index >= 0) && (Index < (int)(ZeCopyCommandQueues.size())),
+            PI_INVALID_VALUE);
+
+  // Return the Ze copy command queue, if already available
+  if (ZeCopyCommandQueues[Index]) {
+    ZeCopyCommandQueue = ZeCopyCommandQueues[Index];
+    return PI_SUCCESS;
+  }
+
+  ZeStruct<ze_command_queue_desc_t> ZeCommandQueueDesc;
+
+  // Ze copy command queue is not avialable at 'Index'. So we create it.
+  if (Index == 0) {
```

Comment:
Why is this the correct check for no queue at "Index"?

Reply:
I meant that the remaining code in this function is used to create a queue.